compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirFunctionParameterChecker.kt

```diff
@@ -70,4 +76,30 @@ object FirFunctionParameterChecker : FirFunctionChecker() {
             }
         }
     }
+
+    private fun checkUninitializedParameter(function: FirFunction<*>, context: CheckerContext, reporter: DiagnosticReporter) {
+        for ((index, parameter) in function.valueParameters.withIndex()) {
+            // Alas, CheckerContext.qualifiedAccesses stack is not available at this point.
+            // Thus, manually visit default value expression and report the diagnostic on qualified accesses of interest.
+            parameter.defaultValue?.accept(object : FirVisitorVoid() {
+                override fun visitElement(element: FirElement) {
+                    element.acceptChildren(this)
+                }
+
+                override fun visitQualifiedAccessExpression(qualifiedAccessExpression: FirQualifiedAccessExpression) {
+                    val namedReference = qualifiedAccessExpression.calleeReference as? FirResolvedNamedReference ?: return
+                    val referredParameter = namedReference.resolvedSymbol.fir as? FirValueParameter ?: return
+                    val referredParameterIndex = function.valueParameters.indexOf(referredParameter)
```

Comment:
I think here you are implicitly relying on `indexOf` returning -1 if it's not in the current function's value parameters, so that in the example:
```kt
fun foo(x: () -> Int = {
    fun bar(xx: () -> Int = { y; 1 }) = xx
    bar()()
}, y: Int) {}
```
...so that when `function == bar, parameter == xx, referredParameter == y`, the index will be -1 (not in `bar`'s parameters and therefore won't be reported there (but it will be when `function == foo, parameter == x`). I'd suggest adding something like `if (index < 0) continue` or adding a comment in here on why it would not be found.

Reply:
Adding an explicit index check sounds better, thanks.